service/controller/resource/monitoring/remotewriteagentconfigsecret/resource.go

```diff
@@ -0,0 +1,117 @@
+package remotewriteagentconfigsecret
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/giantswarm/k8sclient/v7/pkg/k8sclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	pov1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"gopkg.in/yaml.v2"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/prometheus-meta-operator/v2/service/controller/resource/generic"
+	"github.com/giantswarm/prometheus-meta-operator/v2/service/key"
+)
+
+const (
+	Name = "remotewriteagentconfigsecret"
+)
+
+type Config struct {
+	K8sClient  k8sclient.Interface
+	Logger     micrologger.Logger
+	BaseDomain string
+}
+
+func New(config Config) (*generic.Resource, error) {
+	clientFunc := func(namespace string) generic.Interface {
+		c := config.K8sClient.K8sClient().CoreV1().Secrets(namespace)
+		return wrappedClient{client: c}
+	}
+
+	c := generic.Config{
+		ClientFunc:    clientFunc,
+		Logger:        config.Logger,
+		Name:          Name,
+		GetObjectMeta: getObjectMeta,
+		GetDesiredObject: func(ctx context.Context, v interface{}) (metav1.Object, error) {
+			return toSecret(ctx, v, config)
+		},
+		HasChangedFunc: hasChanged,
+	}
+	r, err := generic.New(c)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return r, nil
+}
+
+func getObjectMeta(ctx context.Context, v interface{}) (metav1.ObjectMeta, error) {
+	cluster, err := key.ToCluster(v)
+	if err != nil {
+		return metav1.ObjectMeta{}, microerror.Mask(err)
+	}
+
+	return metav1.ObjectMeta{
```

Comment:
Don't we need to add some labels?

Reply:
Do we really need to add labels like. 
```
clusterID
organization
Installation 
etc? 
```

Reply:
I don't see which other labels to set tbh

Reply:
So i'd rather have suggestions :)